### PR TITLE
feat(liveness): better tests | general code cleanup

### PR DESCRIPTION
**Pull request by danwt**
## Description

- fix storage leak (wasn't deleting old events properly before)
- improves tests for liveness
- liveness wiring is explicitly tested
- improves rapid test too
- removes our old simulation/ junk
- removes several unused methods

----

Closes https://github.com/dymensionxyz/dymension/issues/1253

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Updated the scripts for local run, e.g genesis_config_commands.sh if the PR changes parameters
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

----

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: x/rollapp/keeper/sequencer_hooks.go [hunk 1]
```diff
@@ -20,8 +20,7 @@ func (h SequencerHooks) AfterSetRealProposer(ctx sdk.Context, rollapp string, ne
 	// TODO: tech debt https://github.com/dymensionxyz/dymension/issues/1357
 
 	ra := h.Keeper.MustGetRollapp(ctx, rollapp)
-	h.Keeper.ResetLivenessClock(ctx, &ra)
-	h.Keeper.ScheduleLivenessEvent(ctx, &ra)
+	h.Keeper.IndicateLiveness(ctx, &ra)
 	h.Keeper.SetRollapp(ctx, ra)
```
Comment (danwt) [inline x/rollapp/keeper/sequencer_hooks.go:24]: simplifies
They are equivalent

File: x/rollapp/keeper/liveness.go [hunk 2]
```diff
@@ -59,12 +58,13 @@ func (k Keeper) CheckLiveness(ctx sdk.Context) {
...
 // HandleLivenessEvent will slash or jail and then schedule a new event in the future.
 func (k Keeper) HandleLivenessEvent(ctx sdk.Context, e types.LivenessEvent) error {
-	err := k.sequencerKeeper.SlashLiveness(ctx, e.RollappId)
+	err := k.SequencerK.SlashLiveness(ctx, e.RollappId)
 	if err != nil {
 		return errorsmod.Wrap(err, "slash liveness")
 	}
 
 	ra := k.MustGetRollapp(ctx, e.RollappId)
+	k.DelLivenessEvents(ctx, e.HubHeight, e.RollappId)
```
Comment (danwt) [inline x/rollapp/keeper/liveness.go:67]: @omritoptix fixes a horrible bug which I think would have caused us to slash every block instead of on interval

Comment (omritoptix) [reply x/rollapp/keeper/liveness.go:67]: ha that explains the fast slashing of in the previous simulation!

**Review by omritoptix — APPROVED** [review]

File: x/rollapp/keeper/liveness.go (hunk 2 above)
Comment (danwt) [reply x/rollapp/keeper/liveness.go:67]: yeah sorry had another look but the bug is causing old events to not be deleted (but it doesn't re-read them)
So no impact on users, but it is a storage leak

**Pull request merged**